mlflow/models/signature.py

```diff
@@ -38,6 +39,12 @@
 
 _logger = logging.getLogger(__name__)
 
+_LOG_MODEL_INFER_SIGNATURE_WARNING_TEMPLATE = (
+    "Inferring the model signature from the input example has failed. Reason: %s. Set logging "
+    'level to DEBUG via `logging.getLogger("mlflow").setLevel(logging.DEBUG)` to see the full '
+    "traceback."
```

Comment:
```suggestion
    "Failed to infer the model signature from the input example. Reason: %s. Set logging "
    'level to DEBUG via `logging.getLogger("mlflow").setLevel(logging.DEBUG)` to see the full '
    "traceback."
```

Can we adjust the error message? With this message, users can immediately tell something went wrong.